### PR TITLE
HW Clean Up

### PR DESCRIPTION
**Pull request by derHowie**
Fixes BX-1686
Fixes BX-1661
Fixes BX-1660
Fixes BX-1646

## What changed (plus any additional context for devs)
- [Ensured bridging via common dapps works while using a Trezor Wallet [BX-1686]](https://www.loom.com/share/ac48577fe3bf4be787dfcc126622fd82)
- [We are now preventing an infinite spinner when a user attempts to import multiple indices from a HW device and at least one of the accounts has already been imported. [BX-1661]](https://www.loom.com/share/d45960078beb415e94d8bbe6b7c7d955)
- We corrected a small UI issue reported on the add wallets screen [BX-1660]
- [We are now determining whether a user is signing a dapp request (sign + send) with a hw device prior to the user interacting with the sign CTA](https://www.loom.com/share/9967282a2a194c0fb0680661b60ff115). [We are now also appropriately handling the disconnected ledger flow. Our current implementation results in an infinite spinner](https://www.loom.com/share/823818878aa04f3eb21fe14c49b25705). [BX-1646]

## Screen recordings / screenshots
- [BX-1660] PoW: Screenshot 2024-11-15 at 3 51 18 PM
- [Trezor swap in bx](https://www.loom.com/share/95c6e08f5313424483c8ca840a71b44a)

## What to test
Ensure sign and send are behaving appropriately on Ledger and Trezor. Verify that these changes have not caused any UI issues.

### CODE REVIEW

File: src/entries/popup/pages/messages/SendTransaction/index.tsx [hunk 1]
```diff
@@ -182,6 +183,13 @@ export function SendTransaction({
     return selectedWallet && watchedAddresses?.includes(selectedWallet);
   }, [selectedWallet, watchedWallets]);
 
+  const isSigningWithDevice = useMemo(() => {
+    const signingWithDevice =
+      allWallets.find((w) => isLowerCaseMatch(w.address, selectedWallet))
+        ?.type === 'HardwareWalletKeychain';
+    return signingWithDevice;
+  }, [allWallets, selectedWallet]);
```
Comment (greg-schrammel) [inline src/entries/popup/pages/messages/SendTransaction/index.tsx:191]: we could have something like `const { type, ... } = useWallet(address)`

**Review by brunobar79 — APPROVED** [review]
Good stuff!

**Comment by derHowie** [comment]
@BrodyHughes Can you do a QA pass on this when you have time?

**Comment by derHowie** [comment]
> Tested send / sign on Ledger + Trezor. Also spot checked the PRs mentioned in description. All look good.
> 
> Only one small UI regression found:
> 
> * Wallet has insufficent Eth
> * Go to rainbowkit-example.vercel.app
> * Send Eth
> * Cancel button only takes up small portion of popup width
> * On master it takes up full width
> 
> Screenshot 2024-11-21 at 10 54 15 AM

Fixed in the latest commit:
Screenshot 2024-11-22 at 4 31 20 PM

**Pull request merged**